esmvaltool/diag_scripts/kcs/resampling.py

```diff
@@ -104,11 +164,38 @@ def within_bounds(values, bounds):
     return values.between(low, high)
 
 
+def get_relative_values(top1000, info, period):
+    """Select samples based on the percentile bounds.
+
+    Select samples for which summer pr, and summer and winter tas
+    are within the percentile bounds specified in the recipe.
+    """
+    pr_summer = within_bounds(
+        top1000['pr_summer'], info[f'pr_summer_{period}'])
+    tas_winter = within_bounds(
+        top1000['tas_winter'], info[f'tas_winter_{period}'])
+    tas_summer = within_bounds(
+        top1000['tas_summer'], info[f'tas_summer_{period}'])
+    subset = top1000[
+        np.all([pr_summer, tas_winter, tas_summer], axis=0)
+    ]
+    return subset
+
+
 def determine_penalties(overlap):
     """Determine penalties dependent on the number of overlaps."""
-    return np.piecewise(overlap,
-                        condlist=[overlap < 3, overlap == 3, overlap == 4, overlap > 4],
-                        funclist=[0, 1, 5, 100])
+    return np.piecewise(
```

Comment:
This looks super ugly. Is this required by pylint?

Reply:
autopep made this. I fixed it.